OpenRA.Game/Network/UnitOrders.cs

```diff
@@ -31,7 +32,23 @@ internal static void ProcessOrder(OrderManager orderManager, World world, int cl
 			{
 				// Server message
 				case "Message":
-					TextNotificationsManager.AddSystemLine(order.TargetString);
+					var key = order.TargetString;
+					var arguments = new Dictionary<string, object>();
+					if (order.TargetString.Contains("|"))
```

Comment:
This looks really fragile: what happens if the server wants to send a message (e.g. from motd.txt) that contains literal `|` or `:` characters?

Reply:
I am also not happy with it. Suggestions?

Reply:
Adapt the order format to support what we actually need for this case.

Reply:
One option could be a new "TranslatedString" order that sends a serialized yaml blob that has nodes for the string id and for each of the arguments. There are likely other options, some which might be better than this.

Reply:
I went with a serialized yaml blob.